go-controller/hybrid-overlay/pkg/util/util.go

```diff
@@ -84,14 +84,18 @@ func StartNodeWatch(h types.NodeHandler, wf *factory.WatchFactory) error {
 
 // CopyNamespaceAnnotationsToPod copies annotations from a namespace to a pod
 func CopyNamespaceAnnotationsToPod(k kube.Interface, ns *kapi.Namespace, pod *kapi.Pod) error {
-	nsGw := ns.Annotations[types.HybridOverlayExternalGw]
-	nsVTEP := ns.Annotations[types.HybridOverlayVTEP]
+	nsGw, nsGwExists := ns.Annotations[types.HybridOverlayExternalGw]
```

Comment:
I'm not sure we need the check for existence here.
I *think* the annotator won't write an annotation with an empty value.

Reply:
if types.HybridOverlayExternalGw is not present in the map nsGW is set to "" which the annotator will write